### PR TITLE
docs: fix `ecmaVersion` in one example, add checks

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

While testing https://github.com/eslint/eslint.org/pull/509, I noticed that one example for the `strict` rule has `"ecmaVersion": 6` in the configuration. This works mostly well because `6` is an alias for `2015`, but when the example is opened in the Playground, nothing is selected in the ECMA Version list:

image

#### What changes did you make? (Give an overview)

1. Updated the mentioned configuration to `"ecmaVersion": 2015`.
2. Added a new check in `tools/check-rule-examples.js` - `ecmaVersion`, if present, must be one of 3, 5, 2015, 2016, ..., 2024 (the latest number). `"latest"` string is also disallowed because it is unnecessary since it's the default.

### CODE REVIEW

File: tests/fixtures/good-examples.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ const foo = <bar></bar>;
 
 A test with multiple spaces after 'correct':
 <!-- markdownlint-disable-next-line no-trailing-spaces -->
-:::correct  
+:::correct
```
Comment (mdjermanovic) [inline tests/fixtures/good-examples.md:22]: Looks like this was intentional, so I'll try to revert it.

Comment (mdjermanovic) [reply tests/fixtures/good-examples.md]: Done

**Comment by aladdin-add** [comment]
> but when the example is opened in the Playground, nothing is selected in the ECMA Version list:

Not sure we should update the examples - It seems more of an issue in the playground. Can we fix it in that, I mean, something like normalize the ecmaVersion to year-style, e.g. 6=>2015?

**Comment by mdjermanovic** [comment]
> > but when the example is opened in the Playground, nothing is selected in the ECMA Version list:
> 
> Not sure we should update the examples - It seems more of an issue in the playground. Can we fix it in that, I mean, something like normalize the ecmaVersion to year-style, e.g. 6=>2015?

Serialized configs are generated by our sites, so I think we don't need to support all variants. That said, the old demo (https://archive.eslint.org/demo) was using the 6, 7, 8... style, thus it's a good idea to normalize ecmaVersion in the Playground to support old links 👍 . I'll add normalization in https://github.com/eslint/eslint.org/pull/509. 

But I'd still like to add this check to keep docs' configs consistent. What do you think?

**Comment by aladdin-add** [comment]
for consistency, I'm fine with it - seems it's most used in the docs.

**Review by nzakas — APPROVED** [review]
LGTM. Just leaving open for a couple of days  for others to get a chance to review.

**Comment by fasttime** [comment]
There are 35 rule examples that use `"sourceType": "module"` redundantly. If `"ecmaVersion": "latest"` won't be allowed for consistency reasons, are there any plans to also disallow `"sourceType": "module"` or `"jsx": false`?

**Comment by mdjermanovic** [comment]
> There are 35 rule examples that use `"sourceType": "module"` redundantly. If `"ecmaVersion": "latest"` won't be allowed for consistency reasons, are there any plans to also disallow `"sourceType": "module"` or `"jsx": false`?

Yes, I'm planning to do that in another PR that will update links to send config in the flat format now that https://github.com/eslint/eslint.org/pull/509 is merged.

**Review by fasttime — APPROVED** [review]
LGTM, thanks!

**Pull request merged**